packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2228,10 +2345,10 @@ export class ContainerRuntime
 		localOpMetadata: unknown,
 		local: boolean,
 		expectRuntimeMessageType: boolean,
-	): asserts message is SequencedContainerRuntimeMessage {
+	): asserts message is InboundSequencedContainerRuntimeMessage {
```

Comment:
Just realized this isn't true - if it's a system op this will return early from the default case and this type assertion doesn't hold.
```suggestion
	): void {
```

Reply:
ACTUALLY!  Now that we have the `Inbound` type which includes the `__unknown__` case it seems fair to just type `message: InboundContainerRuntimeMessage` in the function signature.  TS seems to allow it even though at the callsite `message.type` could be any `string`... Not sure why but it works for our case.  This function has to handle the unknown cases.

Reply:
I was thinking similarly.

In all cases (see below):
1. The optimistic destructure should go away.
2. The return should be void.

To simply type message as InboundSequencedContainerRuntimeMessage however, the call-site will require `message as InboundSequencedContainerRuntimeMessage`.
*Almost* alternatively this could accept `InboundSequencedContainerRuntimeMessage | ISequencedDocumentMessage` and the access to `compatDetails` would need guarded by (`'compatDetails' in message`. But then message type going into `compatBehaviorAllowsMessageType` might still be string. Perhaps sync up in the morning.

Reply:
I was able to leverage the callers structure to get to `InboundSequencedContainerRuntimeMessage | ISequencedDocumentMessage` or `InboundSequencedContainerRuntimeMessage` when modern and that let some `as` go away, but it isn't good enough as your `any` comment noted.